.github/workflows/miri.yml

```diff
@@ -0,0 +1,67 @@
+name: Miri linting for move-to-polka
+
+permissions:
+  contents: read
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+
+jobs:
+  miri-test:
+    name: Undefiend behaviour detection for move-to-polka
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        rust: [nightly]
+
+    env:
+      cache_version: 1
+
+    defaults:
+      run:
+        # Set the working directory for all steps (so we only build move-to-polka)
+        working-directory: language/polkavm/move-to-polka
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Cargo registry and target
+        uses: actions/cache@v4
+        with:
+          path: |
+            target
+          key: ${{ env.cache_version }}-${{ runner.os }}-cargo-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ env.cache_version }}-$${{ runner.os }}-cargo-${{ matrix.rust }}-
+
+      - name: Install LLVM 19 on Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt update
+          sudo apt-get install -y llvm-19 llvm-19-dev lld-19 llvm-19-tools clang-19 libclang-19-dev libpolly-19-dev
+          llvm-config-19 --version
+          sudo update-alternatives --install /usr/bin/ld.lld ld.lld /usr/bin/lld-19 100
+
+      - name: Install Rust miri if nightly
+        if: matrix.rust == 'nightly'
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          components: miri
+
+      - name: Build
+        run: cargo build
+
+      - name: Enable unprivileged user namespaces
+        if: matrix.os == 'ubuntu-latest'
+        run: sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
+
+      # - name: Test with Miri
+      #  run: cargo miri test
```

Comment:
shouldn't this be uncommented now?

Reply:
Unfortunately, Miri gives a couple of errors that are too hard for me to correct. So we cannot uncomment this just yet.

Reply:
Thinking about this some more, I think it's fine to enable, even if this workflow fails, at least it brings visibility to the issues. Right now this workflow is just a waste of cycles.